### PR TITLE
[Fix: #45] Added animation to section scrolls [Smooth Scrolling]

### LINKED ISSUE #45: Moving on to sections on the page should be animated
Currently, moving to a section gets the user directly to that section. This should show animated move to the section.
For eg: https://in.pycon.org/2017/#speakers

### PR DESCRIPTION
**Pull request by ashwamegh**
- Adds smooth animation while scrolling to sections and Fixes #45 
- Changes can be reviewed at https://shashank7200.github.io/inpycon2017/

### CODE REVIEW

**Comment by realslimshanky** [comment]
@shashank7200 here is the problem with your implementation of smooth scroll.
1) It only works on the index.html i.e. the homepage of in.pycon.org
2) For every other page the link for speakers is dead i.e. clicking on 'speakers' does nothing/ takes you nowhere.
You can try to make it work by applying smooth scroll only on the index.html.

File: js/main.js [hunk 1]
```diff
@@ -48,6 +48,15 @@ jQuery(document).on('ready', function() {
 			scrollTop: jQuery("#tg-main").offset().top + offset
 		}, 2000);
 	});
+
+	/* -------------------------------------
+	     ADD ANIMATIONS TO SECTION SCROLL
+	 -------------------------------------- */
+    $('a[href*=\\#]').on('click', function(event){
+        event.preventDefault();
```
Comment (realslimshanky) [inline js/main.js]: This makes 'Speakers' button a dead link when clicked from any other page than index.html. Different approach is needed to make it work.

**Review by realslimshanky — COMMENTED** [review]
Trying to make smooth scroll work like this is leaving dead links behind. A different approach is needed to make this work.

File: js/main.js (hunk 1 above)
Comment (ashwamegh) [reply js/main.js]: @realslimshanky  okay, let me fix that.

**Comment by realslimshanky** [comment]
@shashank7200 according to your new changes, on clicking on 'Speakers' takes us to the Speaker tab on index.html page but smooth scroll is not working. i.e. the feature that you are proposing works only when we are at index.html and not on any other page (which is 6 out of 7 case).

**Comment by ashwamegh** [comment]
@realslimshanky solved it!

**Comment by realslimshanky** [comment]
There are few glitches on the mobile browsers while testing. I would wait to see other people's review.

File: js/main.js [hunk 2]
```diff
@@ -48,6 +48,32 @@ jQuery(document).on('ready', function() {
 			scrollTop: jQuery("#tg-main").offset().top + offset
 		}, 2000);
 	});
+
+	/* -------------------------------------
+	     ADD ANIMATIONS TO SECTION SCROLL
+	 -------------------------------------- */
+    function ea_scroll( hash ) {
```
Comment (CuriousLearner) [inline js/main.js]: Please indent your JS correctly :)

**Comment by CuriousLearner** [comment]
@shashank7200 one level backward. Should go at the same level as your comment.

**Comment by ashwamegh** [comment]
@CuriousLearner  When I indented the code in the editor, it was perfect but, after pushing code to github, it is showing unindented.

**Comment by ashwamegh** [comment]
@CuriousLearner done!

**Comment by CuriousLearner** [comment]
Hey,

Thanks for your contributions!

As I see on the link you mentioned, whenever I click on the #speakers link as you mentioned on the issue, it redirects me to your portfolio site.

Can you please have a look at this?

**Comment by ashwamegh** [comment]
@CuriousLearner See that is not a problem, that is a behaviour of that website which was earlier implemented, I can change it but, it'll be not good to change it. You can test it locally, it'll work fine. I had long discussion with @realslimshanky explaining this behaviour. From a talk to @realslimshanky , what I explained to him:
1. See, you can expect speakers section to be available on my github https://shashank7200.github.io/
2. The speakers link is implemented as href=/#speakers, so it will always point to your home directly and then search for speakers  section
3. Github creates a route for every repo other than your own github page i.e. in my case https://shashank7200.github.io/ {i.e. home route]
and my fork repo of python website is  is at https://shashank7200.github.io/inpycon2017 which is not acting as the home directory
4. That's why either you can review it locally or merge the pull request, it'll automatically be available on the website.

**Comment by CuriousLearner** [comment]
I know this would show up correctly once deployed or on local testing. My main concern here is, when we archive this site, it should work correctly that time too, which not might be the case.

**Comment by ashwamegh** [comment]
@CuriousLearner In this case, what should I do then?

**Comment by CuriousLearner** [comment]
@shashank7200 Rather than referencing it from the root, add an anchor with respect to the current page.

**Comment by ashwamegh** [comment]
@CuriousLearner I am doing what you suggested, but I must say, it will not redirect you to the homepage's speakers section afterwards from external links

**Comment by CuriousLearner** [comment]
Yes, you're right. Let me think how can we do this.

**Comment by ashwamegh** [comment]
Okay @CuriousLearner

**Comment by realslimshanky** [comment]
Sorry for delay @shashank7200 
After reviewing current change here are my concerns.
* You need to change the `href` property on `a` tag of speakers from `#speakers` to `/2017/#speakers`. In this case even after archiving the links will work.
* Then I'm afraid that we'll be back to square 1 when the smooth scroll wasn't working 6 out of 7 times.
Try this fix and let me know.

**Comment by ashwamegh** [comment]
@realslimshanky  Done changes locally, but it breaks the functionality. I can suggest we can do the needful changes while archiving and leave the `href` as `/#speakers` or `#speakers` for now. Should I push these changes to remote?

Here's the output of locally tested changes:

Local inpycon test

**Comment by realslimshanky** [comment]
This is temporary but might work.
What do you think @CuriousLearner ?
Should we go for temporary fix and change the links before archiving next year
or the permanent one and test it after merging?

**Comment by CuriousLearner** [comment]
I'm not sure.

@theskumar How should we go around this/

**Pull request closed without merging**